packages/babel-parser/test/fixtures/experimental/discard-binding/invalid-array-expression/output.json

```diff
@@ -0,0 +1,230 @@
+{
+  "type": "File",
+  "start":0,"end":108,"loc":{"start":{"line":1,"column":0,"index":0},"end":{"line":9,"column":21,"index":108}},
+  "errors": [
+    "SyntaxError: Unexpected void binding. (1:4)",
```

Comment:
I would prefer something like `When not used in a binding position, "void" must be followed by an expression.`

Reply:
On a second thought, I extracted a new error type for the suggested message when CoverDiscardElement is seen. So that we still throw "unexpected void binding" on cases where `void` is indeed a binding but not allowed in the spec:
```js
var void = initializer;
```

https://github.com/babel/babel/pull/17163/commits/f8d042b492a55fc70e85e5b16cc56ddd96ac367a